src/Data/Functor/Rep.hs

```diff
@@ -432,22 +438,40 @@ tabulateCotraverse1 =
 indexLogarithm :: f a -> Logarithm f -> a
 indexLogarithm = flip runLogarithm
 
--- | Derive 'cotraverse1' via an isomorphism
-cotraverse1Iso ::
-     (Representable g, Functor1 w)
-  => (forall x. f x -> g x)
-  -> (forall x. g x -> f x)
-  -> (w Identity -> a)
-  -> w f
-  -> f a
-cotraverse1Iso t frm f = frm . cotraverseMap1 f t
-
-gcotraverse1 ::
+cotraverseMap1Iso ::
+     (Representable f', Functor1 w)
+  => (forall x. f x -> f' x)
+  -> (f' a -> f a)
+  -> (w Identity -> a) -> (forall x. g x -> f x) -> w g -> f a
+cotraverseMap1Iso too frm f g = frm . cotraverseMap1 f (too . g)
+
+#if __GLASGOW_HASKELL__ >= 806
+-- Yes, those constraints sure do look redundant, but QuantifiedConstraints
+-- is pretty flaky with coercibility.
```

Comment:
Blargle.... This takes too many arguments. I guess we need to make an `Internal` module so we can add an unsafe version to use for older GHC versions.